packages/drivers/odsp-driver/src/odspDelayLoadedDeltaStream.ts

```diff
@@ -225,8 +226,28 @@ export class OdspDelayLoadedDeltaStream {
 		});
 	}
 
-	private async joinSession(requestSocketToken: boolean, options: TokenFetchOptionsEx) {
-		const response = await this.joinSessionCore(requestSocketToken, options).catch((e) => {
+	private async joinSession(
+		requestSocketToken: boolean,
+		options: TokenFetchOptionsEx,
+		isRefreshingJoinSession: boolean,
+	) {
+		// If this call is to refresh the join session for the current connection but we are already disconnected in
+		// the meantime then do not make the call. However, we should not have come here if that is the case because
+		// timer should have been disposed, but due to race condition with the timer we should not make the call and
+		// throw error.
+		if (isRefreshingJoinSession && this.currentConnection === undefined) {
```

Comment:
Can we do even more targetted check here? I.e. not just that we have this.currentConnection, but that we have exactly same connection object we started with? Maybe by passing in clientId into here (i.e. remembering it when scheduling timer) and checking it here.
There is good chance we have a reconnect and somehow missed cancellation, and then we might have 2 sets of calls to joinSession :)